mesonbuild/interpreterbase.py

```diff
@@ -55,6 +55,37 @@ def wrapped(self, node, args, kwargs):
         return f(self, node, args, kwargs)
     return wrapped
 
+#This decorator does the same than permittedKwargs plus:
+# - checks for file extensions
+# - converts str values to File
+class processArgs:
+    def __init__(self, extensions_dict):
+        self.extensions_dict = extensions_dict
+
+    def __call__(self, f):
+        @wraps(f)
+        def wrapped(s, node_or_state, args, kwargs):
+            if hasattr(s, 'subdir'):
+                subdir = s.subdir
+                lineno = s.current_lineno
+                source_dir = s.environment.source_dir
+            elif hasattr(node_or_state, 'subdir'):
+                subdir = node_or_state.subdir
+                lineno = node_or_state.current_lineno
+                source_dir = node_or_state.environment.source_dir
+            for key, item in kwargs.items():
+                if key in self.extensions_dict:
+                    extensions = self.extensions_dict[key]
+                    if not extensions is None:
```

Comment:
__[Flake8]__

test for object identity should be 'is not'


<!-- Comment by [SideCI](https://sideci.com) -->